public/app/features/dashboard/state/PanelQueryRunner.ts

```diff
@@ -64,20 +67,29 @@ export class PanelQueryRunner {
    * Returns an observable that subscribes to the shared multi-cast subject (that reply last result).
    */
   getData(transform = true): Observable<PanelData> {
-    if (transform) {
-      return this.subject.pipe(
-        map((data: PanelData) => {
-          if (this.hasTransformations()) {
-            const newSeries = transformDataFrame(this.transformations, data.series);
-            return { ...data, series: newSeries };
-          }
-          return data;
-        })
-      );
-    }
-
-    // Just pass it directly
-    return this.subject.pipe();
+    return this.subject.pipe(
+      // apply overrides
+      map((data: PanelData) => {
+        if (this.fieldOverrideOptions) {
+          const series = applyFieldOverrides({ data: data.series, ...this.fieldOverrideOptions });
+          return {
+            ...data,
+            series,
+          };
+        }
+        return data;
+      }),
+      // apply transformations
+      map((data: PanelData) => {
+        if (transform && this.hasTransformations()) {
+          return {
+            ...data,
+            series: transformDataFrame(this.transformations, data.series),
+          };
+        }
+        return data;
```

Comment:
Is there a good reason to separate these into two separate map functions?  Seems like they could easily be part of the same map function and just comments that separate them. 

Or something like 

```typescript
return this.subject.pipe(
   map(this.applyTransformations),  
   map(this.applyFieldOverrides)
); 
```

Btw I think transformations need to go before field overrides as transformations can filter out or add new fields.

Reply:
Totally right!